util/collections.go

```diff
@@ -21,3 +21,31 @@ func RemoveElementFromList(list []string, element string) []string {
 	}
 	return out
 }
+
+// Returns a copy of the given list with all duplicates removed (keeping the first encountereds)
+func RemoveDuplicatesFromList(list []string) []string {
+	return removeDuplicatesFromList(list, false)
+}
+
+// Returns a copy of the given list with all duplicates removed (keeping the last encountereds)
+func RemoveDuplicatesFromListKeepLast(list []string) []string {
+	return removeDuplicatesFromList(list, true)
+}
+
+func removeDuplicatesFromList(list []string, keepLast bool) []string {
+	out := make([]string, 0, len(list))
+	present := make(map[string]bool)
+
+	for _, value := range list {
+		if _, ok := present[value]; ok {
+			if keepLast {
+				out = RemoveElementFromList(out, value)
+			} else {
+				continue
+			}
+		}
+		out = append(out, value)
+		present[value] = true
+	}
```

Comment:
It seems like a simpler way to implement this (and a faster way, though that doesn't matter here) is to load the map first:

```
// pseudo code
for value in list:
  if (value not in present) or (value in present and keepLast):
    present[value] = true
```

And then you just return all the keys in the `present` map.